### PR TITLE
T-3 Sign Up page

### PR DESCRIPTION
**Pull request by VladyslavGilyuk**
Sign Up page:
SingUpFormHelper (new Fields: First/Last name, Password Confirmation).
Password Confirmation logic.
Custom useHook - useSignUpForm.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
     ]
   },
   "devDependencies": {
+    "@types/react-helmet": "^6.1.6",
```
Comment (mazhula97) [inline package.json]: Again

File: src/components/common/formInput/index.tsx [hunk 2]
```diff
@@ -42,7 +46,13 @@ const FormInput: React.FC<FormInputProps> = ({
         fullWidth={true}
         type={type === 'password' && !showPassword ? 'password' : 'text'}
         error={!!errors[name]}
-        helperText={errors[name]?.message ?? ' '}
+        helperText={
```
Comment (mazhula97) [inline src/components/common/formInput/index.tsx]: What is the reason to change this prop? it should be the same

File: src/components/form/singUp/index.tsx [hunk 3]
```diff
@@ -1,5 +1,66 @@
...
+      <form onSubmit={handleSubmit(onSubmit)}>
+        {SingUpFormHelper.map(({ name, label, validations, placeholder, type }) => (
+          <FormInput
+            key={name}
+            name={name}
+            label={label}
+            validations={validations}
+            placeholder={placeholder}
+            type={type}
+            showPassword={showPassword}
```
Comment (mazhula97) [inline src/components/form/singUp/index.tsx]: It is not a reusable component anymore
FormInput knows a lot of unnecessary info.
password, showPassord, handleClickShowPassword, handleMouseDownPassword, passwordConfirmation
To many specific props

File: src/hooks/useSignUpForm.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  const handleMouseDownPassword = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+  };
+
+  const {
+    handleSubmit,
+    register,
+    watch,
+    formState: { errors },
+  } = useForm<ISignUp>();
```
Comment (mazhula97) [inline src/hooks/useSignUpForm.tsx]: image

don't see any diff. Please move 
`const [showPassword, setShowPassword] = useState(false);
  const handleClickShowPassword = () => setShowPassword((show) => !show);

  const handleMouseDownPassword = (event: React.MouseEvent<HTMLButtonElement>) => {
    event.preventDefault();
  };

  const {
    handleSubmit,
    register,
    watch,
    formState: { errors },
  } = useForm<ISignUp>();`
 into useAuthForm(hooks folder) use it in useSignInForm and useSignUpForm. And move useSignInForm & useSignUpForm in to the folders with components

**Review by mazhula97 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (VladyslavGilyuk) [reply package.json]: Removed all helmet dependencies

File: src/components/common/formInput/index.tsx (hunk 2 above)
Comment (VladyslavGilyuk) [reply src/components/common/formInput/index.tsx]: The reason is to check if we have a ConfirmPassword input if yes then check if it is same as password  (password and confirmPassword variables are not stored in FormHelpers and we cannot map them from there (at least of my knowledge) and therefore the check is handled here).

File: src/hooks/useSignUpForm.tsx (hunk 4 above)
Comment (VladyslavGilyuk) [reply src/hooks/useSignUpForm.tsx]: useAuthForm created in hooks and used in useSignInForm and useSignUpForm hooks which are moved to components=>form=>signIn/signUp (right?).

File: src/components/form/singUp/index.tsx (hunk 3 above)
Comment (VladyslavGilyuk) [reply src/components/form/singUp/index.tsx]: I assume I am supposed to return it to reusable component again but I am not sure how exactly to do that.

File: src/utils/formHelpers.ts
Comment (mazhula97) [inline src/utils/formHelpers.ts:1]: ISingInFormHelper can't contain firstName' | 'lastName' | 'passwordConfirmation' name.
Create generic instance like IBaseFieldType and add necessary fields

File: src/components/form/singUp/index.tsx [hunk 5]
```diff
@@ -1,5 +1,66 @@
...
+            handleClickShowPassword={handleClickShowPassword}
```
Comment (mazhula97) [inline src/components/form/singUp/index.tsx]: Next 2 props are unused

File: src/components/form/singUp/index.tsx [hunk 6]
```diff
@@ -1,5 +1,66 @@
...
+import { SingUpFormHelper } from '../../../utils/formHelpers';
+import useSignUpForm from './useSignUpForm';
+import { FormInputWrapper, StyledLoginButton } from './styled';
+
+export interface ISignUp {
+  email?: string;
+  password?: string;
+  passwordConfirmation?: string;
+}
+export interface FieldErrors {
```
Comment (mazhula97) [inline src/components/form/singUp/index.tsx]: Changed. Look at DM

File: src/components/common/formInput/index.tsx (hunk 2 above)
Comment (mazhula97) [reply src/components/common/formInput/index.tsx]: This should be a reusable field that shouldn't know about input names. (except of some corner cases).
P.S. Patch in DM

**Review by mazhula97 — APPROVED** [review]

File: src/hooks/useAuthForm.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  email: string;
+  password: string;
+}
+interface ISignUp extends ISignIn {
+  firstname: string;
+  lastname: string;
+  passwordConfirmation: string;
+}
+
+export interface CommonFieldValues extends FieldValues {
```
Comment (mazhula97) [inline src/hooks/useAuthForm.tsx:13]: Always Use "I" as a first letter for interfaces. 
It shouldn't works in such way, we don't have passwordConfirmation or firstname fields in signIn model.
putting it into tech debt

Comment (VladyslavGilyuk) [reply src/hooks/useAuthForm.tsx:13]: Noted down

**Pull request merged**